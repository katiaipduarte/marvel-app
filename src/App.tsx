import logo from './assets/marvel_logo.png';
import Footer from './components/Footer/Footer';
import ComicsList from './components/ComicsList/ComicsList';
import SearchBar from './components/SearchBar/SearchBar';
import { useState } from 'react';

const App = (): JSX.Element => {
  const [characterId, setCharacterId] = useState<number | null>(null);

  return (
    <>
      <header>
        <img src={logo} className="logo" alt="marvel logo" />
        <SearchBar find={(id: number | null) => setCharacterId(id)} />
      </header>
      <main>
        <ComicsList characterId={characterId} />
      </main>
      <Footer />
    </>
  );
};

export default App;
