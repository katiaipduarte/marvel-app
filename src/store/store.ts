import { combineReducers, applyMiddleware, createStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import favouritesReducer from './favourites/reducer';
import { Favourites } from './favourites/type';

export interface GlobalState {
  favouritesState: Favourites;
}

const combinedReducer = combineReducers<GlobalState>({
  favouritesState: favouritesReducer,
});

const bindMiddleware = (middleware: any) => {
  if (process.env.NODE_ENV !== 'production') {
    const { composeWithDevTools } = require('redux-devtools-extension');
    return composeWithDevTools(applyMiddleware(...middleware));
  }
  return applyMiddleware(...middleware);
};

const store = createStore(combinedReducer, bindMiddleware([thunk]));

export default store;
