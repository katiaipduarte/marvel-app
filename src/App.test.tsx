import { render, screen } from '@testing-library/react';
import App from './App';
import store from './store/store';
import { Provider } from 'react-redux';
import axios from './lib/services/__mocks__/axios';

describe('the <App /> component', () => {
  afterEach(() => {
    axios.get.mockClear();
  });

  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders logo element', async () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const logoElement = screen.getByAltText('marvel logo');
    expect(logoElement).toBeInTheDocument();
  });
});
