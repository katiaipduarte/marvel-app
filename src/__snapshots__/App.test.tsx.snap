// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <App /> component should render correctly 1`] = `
<header>
  <img
    alt="marvel logo"
    class="logo"
    src="marvel_logo.png"
  />
  <form
    aria-label="Search comics by character form"
    class="sc-iCoGMd kDgDSP"
    role="searchbox"
    tabindex="-1"
  >
    <button
      aria-label="Click to search"
      id="search-term-btn"
      tabindex="-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </button>
    <input
      aria-label="Search comics by character"
      maxlength="50"
      placeholder="Search comics by character"
      role="search"
      type="text"
      value=""
    />
    <button
      aria-label="Click to clear search"
      id="clear-search-term-btn"
      style="visibility: hidden;"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
        />
      </svg>
    </button>
  </form>
</header>
`;
